src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+    private double hypotenuse;
+
+    public RightTriangle(String color, double firstLeg, double secondLeg, double hypotenuse) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+        this.hypotenuse = hypotenuse;
+    }
+
+    @Override
+    public double getArea() {
+        return (firstLeg * secondLeg) / 2;
+    }
+
+    @Override
+    public void draw() {
+        String randomColor = new ColorSupplier().getRandomColor();
```

Comment:
with the current implementation, you instantiate class ColorSupplier each time the method is being invoked.

think about instantiating ColorSupplier once and reusing it multiple times.